src/SelfRoleManager.ts

```diff
@@ -255,10 +259,15 @@ export class SelfRoleManager extends EventEmitter {
     const userWantsToRemoveRole = isButtonInteraction ? memberHasRole : memberHasRole && ((!rteData.removeOnReact && isReactionRemoval) || (rteData.removeOnReact && !isReactionRemoval));
     const userWantsToAddRole = isButtonInteraction ? !memberHasRole : !memberHasRole && ((!rteData.removeOnReact && !isReactionRemoval) || (rteData.removeOnReact && isReactionRemoval));
     const role: Role = rteData.role instanceof Role ? rteData.role : await userAction.message.guild.roles.fetch(rteData.role);
+    const userNotMetDependencies = rteData.dependencies?.some((role: RoleResolvable) => memberRoles.find((memberRole: Role) => (role instanceof Role ? role === memberRole : role === memberRole.id)) === undefined) ?? false;
```

Comment:
Apart from the alignment around the "dependencies" word, I understand that you probably want to optimize the comparison between each array's content.
However, the readability is also important as this code will be maintained by me. I cannot stress enough the fact that the chained operations unfortunately do not reflect a proper English sentence...

Here are my pieces of advice:
- do not strictly compare a `.find()` result to a `undefined`.
- rather than inversing the condition for performance, question yourself on what you try to achieve (you want to know if every required roles are in the member roles' list or not, right?).
- use some indentation to ease the readibility of the whole line.

All in all, a proper assignation of such variable would be:
```suggestion
    const userMeetsDependencies = rteData.dependencies
      ?.every(role => role instanceof Role
        ? memberRoles.includes(role)
        : memberRoles.map(r => r.id).includes(role)
      ) ?? true;
```